### ISSUE TITLE
How to predict from RPN model?

### ISSUE THREAD

**Issue by mmiisskk** [broadinstitute/keras-rcnn#136]
Hi,
please, what is the right way to use predict from RPN model when I am loading saved weights and new image to predict? 

I try to use your example in '/models/_rpn.py' and after ran 'model.fit_generator' I saved weights using 'model.save_weights()' from keras.
Now I would like to load these saved weights and run 'model.predict' on new image which wasn't in training process - but it give me the error : 
> "ValueError: The model expects 4  arrays, but only received one array. Found: array with shape (303, 540, 3)"

And maybe this is because RPN model is expecting the input with bounding_boxes, labels, image and metadata but I am giving just image to it.

How can I change the RPN model to get right mode just for prediction for objects in new image?
Really thanks!

**Comment by jhung0** [comment]
Yes, the input should be bounding_boxes, labels, image, metadata

**Comment by mmiisskk** [comment]
Hm, that's the problem, I think. 
Because when I have new image for prediction there are no bounding_boxes, labels and metadata - there is just image and I'd like to get these information (b_boxes, labels) from prediction. 
Can I achieve it?

**Comment by LSBigum** [comment]
Did you find a solution to this?

**Comment by mmiisskk** [comment]
> Did you find a solution to this?

No, I studied keras-retinanet instead.
